TechItEasyDatabase_GW.sql

```diff
@@ -0,0 +1,125 @@
+DROP TABLE IF EXISTS Users;
+DROP TABLE IF EXISTS TelevisionCIModules;
+DROP TABLE IF EXISTS TelevisionWallBrackets;
+DROP TABLE IF EXISTS Televisions;
+DROP TABLE IF EXISTS RemoteControllers;
+DROP TABLE IF EXISTS CIModules;
+DROP TABLE IF EXISTS WallBrackets;
+
+
+CREATE TABLE Users
+(
+	username VARCHAR(50) NOT NULL UNIQUE,
+	password VARCHAR(50) NOT NULL
+);
+
+CREATE TABLE RemoteControllers
+(
+-- Attributen vanuit de abstracte Product klasse
+	productid INT PRIMARY KEY,
+	productname VARCHAR(50) NOT NULL,
+	brand VARCHAR(50),
+	price DOUBLE PRECISION,
+	currentStock INT NOT NULL,
+	sold INT NOT NULL,
+-- /Attributen vanuit de abstracte Product klasse
+	compatibleWith VARCHAR(50),
+	batteryType VARCHAR(50)
+);
+
+CREATE TABLE Televisions
+(
+-- Attributen vanuit de abstracte Product klasse
+	productid INT PRIMARY KEY,
+	productname VARCHAR(50) NOT NULL,
+	brand VARCHAR(50),
+	price DOUBLE PRECISION,
+	currentStock INT NOT NULL,
+	sold INT NOT NULL,
+-- /Attributen vanuit de abstracte Product klasse
+	televisionType VARCHAR(50) NOT NULL,
+	available DOUBLE PRECISION,
+	refreshRate DOUBLE PRECISION,
+	screenType VARCHAR(50),
+	remoteControlID INT,
+	CONSTRAINT FK_Televisions_Remotecontrollers
+	FOREIGN KEY (remoteControlID) REFERENCES remoteControllers(productid)
+);
+
+CREATE TABLE CIModules
+(
+-- Attributen vanuit de abstracte Product klasse
+	productid INT PRIMARY KEY,
+	productname VARCHAR(50) NOT NULL,
+	brand VARCHAR(50),
+	price DOUBLE PRECISION,
+	currentStock INT NOT NULL,
+	sold INT NOT NULL
+-- /Attributen vanuit de abstracte Product klasse
+);
+
+-- Koppeltabel: Televisions & CI Modules
+CREATE TABLE TelevisionCIModules
+(
+	id SERIAL PRIMARY KEY,
+	televisionID INT NOT NULL,
+	cimoduleID INT NOT NULL,
+	CONSTRAINT FK_TelevisionCIModules_Televisions
+	FOREIGN KEY (televisionID) REFERENCES Televisions(productID),
+	CONSTRAINT FK_TelevisionCIModules_CIModules
+	FOREIGN KEY (cimoduleID) REFERENCES CIModules(productID)
+);
+
+CREATE TABLE WallBrackets
+(
+-- Attributen vanuit de abstracte Product klasse
+	productid INT PRIMARY KEY,
+	productname VARCHAR(50) NOT NULL,
+	brand VARCHAR(50),
+	price DOUBLE PRECISION,
+	currentStock INT NOT NULL,
+	sold INT NOT NULL,
+-- /Attributen vanuit de abstracte Product klasse
+	adjustable BOOLEAN
+);
+
+-- Koppeltabel: Televisions & WallBrackets
+CREATE TABLE TelevisionWallBrackets
+(
+	id SERIAL PRIMARY KEY,
+	televisionID INT NOT NULL,
+	wallbracketID INT NOT NULL,
+	CONSTRAINT FK_TelevisionWallBrackets_Televisions
+	FOREIGN KEY (televisionID) REFERENCES Televisions(productID),
+	CONSTRAINT FK_TelevisionWallBrackets_WallBrackets
+	FOREIGN KEY (wallbracketID) REFERENCES WallBrackets(productID)
+);
+
+INSERT INTO Users (Username, Password) VALUES ('Name of user', 'PassKey1'), ('Name of 2nd user', 'PassKey2');
```

Comment:
het viel me op dat hier niet alle kolommen een waarde krijgen, bijvoorbeeld price in televisions